packages/framework-provider-aws/src/library/events-adapter.ts

```diff
@@ -110,3 +110,55 @@ export async function storeEvents(
   await dynamoDB.batchWrite(params).promise()
   logger.debug('[EventsAdapter#storeEvents] EventEnvelope stored')
 }
+
+export async function destroyEntity(
+  dynamoDB: DynamoDB.DocumentClient,
+  config: BoosterConfig,
+  logger: Logger,
+  entityTypeName: string,
+  entityID: UUID
+): Promise<void> {
+  logger.debug('[EventsAdapter#destroyEntity] Destroying entity')
+
+  // We need to fetch the entity events and snapshots
+  const events = await readEntityEventsSince(dynamoDB, config, logger, entityTypeName, entityID)
+  const snapshots = await readEntityEventsSince(
+    dynamoDB,
+    config,
+    logger,
+    entityTypeName,
+    entityID,
+    originOfTime,
+    'snapshot'
+  )
+
+  if (events) {
+    let eventsToDelete = [...events]
+    if (snapshots) {
+      eventsToDelete = [...eventsToDelete, ...snapshots]
+    } else {
+      logger.debug(`[EventsAdapter#destroyEntity] No snapshots found for entity ${entityTypeName} with id: ${entityID}`)
+    }
```

Comment:
This is a matter of personal preference, but perhaps you like the same version as me 😃, with less indentation

```suggestion
    if (!snapshots) {
      logger.debug(`[EventsAdapter#destroyEntity] No snapshots found for entity ${entityTypeName} with id: ${entityID}`)
    }
    const eventsToDelete = [...events, ...snapshots ?? []]
```